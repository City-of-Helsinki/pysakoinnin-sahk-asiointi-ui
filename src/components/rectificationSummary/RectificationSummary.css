.rectification-summary-container {
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.rectification-summary-subgrid {
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.rectification-summary-content, .rectification-summary-content div, .rectification-summary-content textarea {
  height: 100% !important;
}

.rectification-summary-header {
  margin-top: var(--spacing-xl);
  font-weight: 500;
  line-height: 32px;
  letter-spacing: -0.2px;
}

.rectification-summary-fine-details {
  margin-top: var(--spacing-m);
  grid-column: 1 / 3;
}

.rectification-summary-details > div {
  margin-bottom: 24px;
}
