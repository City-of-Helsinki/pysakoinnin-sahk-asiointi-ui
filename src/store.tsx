import { configureStore } from '@reduxjs/toolkit';
import formContentReducer from './components/formContent/formContentSlice';
import formStepperReducer from './components/formStepper/formStepperSlice';
import extendDueDateFormReducer from './components/extendDueDate/extendDueDateFormSlice';
import userReducer from './components/user/userSlice';

const store = configureStore({
  reducer: {
    formContent: formContentReducer,
    formStepper: formStepperReducer,
    extendDueDateForm: extendDueDateFormReducer,
    user: userReducer
  }
});

export default store;
export type RootState = ReturnType<typeof store.getState>;
