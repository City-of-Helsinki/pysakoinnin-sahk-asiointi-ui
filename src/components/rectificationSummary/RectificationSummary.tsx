import React from 'react';
import { useSelector } from 'react-redux';
import { Accordion, TextArea, TextInput } from 'hds-react';

import './RectificationSummary.css';
import { useTranslation } from 'react-i18next';
import InfoContainer from '../infoContainer/InfoContainer';
import { selectFormContent } from '../formContent/formContentSlice';

const RectificationSummary = () => {
  const { t } = useTranslation();
  const selectedForm = useSelector(selectFormContent).selectedForm;

  return (
    <>
      <h2 className="rectification-summary-header">
        {t('rectification:rectification-info')}
      </h2>
      <div className="rectification-summary-container">
        <div className="rectification-summary-details">
          <TextInput
            id="relation"
            label={t(`rectification:relation-info:${selectedForm}:relation`)}
            value={t(`rectification:relation-info:${selectedForm}:driver`)}
            readOnly
          />
          <TextInput
            id="name"
            label={t('common:name')}
            value="Etunimi Sukunimi"
            readOnly
          />
          <TextInput
            id="ssn"
            label={t('common:ssn')}
            value="123456-123A"
            readOnly
          />
          <TextInput
            id="rectification-address"
            label={t('rectification:address')}
            value="Elimäenkatu 5"
            readOnly
          />
          <div className="rectification-summary-subgrid">
            <TextInput
              id="zipcode"
              label={t('rectification:zipcode')}
              value="00100"
              readOnly
            />
            <TextInput
              id="city"
              label={t('rectification:city')}
              value="Helsinki"
              readOnly
            />
          </div>
          <TextInput
            id="email"
            label={t('common:email')}
            value="etunimi@email.com"
            readOnly
          />
          <TextInput
            id="phone"
            label={t('common:phone')}
            value="+358401234567"
            readOnly
          />
          <TextInput
            id="IBAN"
            label={t('rectification:IBAN')}
            value="FI9780001700903330"
            readOnly
          />
          <TextInput
            id="decision"
            label={t('rectification:decision-choice')}
            value="Pysäköinnin asiointikansiooni"
            readOnly
          />
        </div>
        <div className="rectification-summary-content">
          <TextArea
            readOnly
            id="rectification-content"
            label={t('rectification:rectification-content')}
            value="Mieleni minun tekevi, aivoni ajattelevi lähteäni laulamahan, saa'ani sanelemahan, sukuvirttä suoltamahan, lajivirttä laulamahan. Sanat suussani sulavat, puhe'et putoelevat, kielelleni kerkiävät, hampahilleni hajoovat.Veli kulta, veikkoseni, kaunis kasvinkumppalini! Lähe nyt kanssa laulamahan, saa kera sanelemahan yhtehen yhyttyämme, kahta'alta käytyämme! Harvoin yhtehen yhymme, saamme toinen toisihimme näillä raukoilla rajoilla, poloisilla Pohjan mailla.Lyökämme käsi kätehen, sormet sormien lomahan, lauloaksemme hyviä, parahia pannaksemme, kuulla noien kultaisien, tietä mielitehtoisien, nuorisossa nousevassa, kansassa kasuavassa: noita saamia sanoja, virsiä virittämiä vyöltä vanhan Väinämöisen, alta ahjon Ilmarisen, päästä kalvan Kaukomielen, Joukahaisen jousen tiestä, Pohjan peltojen periltä, Kalevalan kankahilta.Niit' ennen isoni lauloi kirvesvartta vuollessansa; niitä äitini opetti väätessänsä värttinätä, minun lasna lattialla eessä polven pyöriessä, maitopartana pahaisna, piimäsuuna pikkaraisna. Sampo ei puuttunut sanoja eikä Louhi luottehia: vanheni sanoihin sampo, katoi Louhi luottehisin, virsihin Vipunen kuoli, Lemminkäinen leikkilöihin.Viel' on muitaki sanoja, ongelmoita oppimia: tieohesta tempomia, kanervoista katkomia, risukoista riipomia, vesoista vetelemiä, päästä heinän hieromia, raitiolta ratkomia, paimenessa käyessäni, lasna karjanlaitumilla, metisillä mättähillä, kultaisilla kunnahilla, mustan Muurikin jälessä, Kimmon kirjavan keralla.Vilu mulle virttä virkkoi, sae saatteli runoja. Virttä toista tuulet toivat, meren aaltoset ajoivat. Linnut liitteli sanoja, puien latvat lausehia. Mieleni minun tekevi, aivoni ajattelevi lähteäni laulamahan, saa'ani sanelemahan, sukuvirttä suoltamahan, lajivirttä laulamahan. Sanat suussani sulavat, puhe'et putoelevat, kielelleni kerkiävät, hampahilleni hajoovat.Veli kulta, veikkoseni, kaunis kasvinkumppalini! Lähe nyt kanssa laulamahan, saa kera sanelemahan yhtehen yhyttyämme, kahta'alta käytyämme! Harvoin yhtehen yhymme, saamme toinen toisihimme näillä raukoilla rajoilla, poloisilla Pohjan mailla.Lyökämme käsi kätehen, sormet sormien lomahan, lauloaksemme hyviä, parahia pannaksemme, kuulla noien kultaisien, tietä mielitehtoisien, nuorisossa nousevassa, kansassa kasuavassa: noita saamia sanoja, virsiä virittämiä vyöltä vanhan Väinämöisen, alta ahjon Ilmarisen, päästä kalvan Kaukomielen 2374 merkkiä"
          />
        </div>

        <Accordion
          id="summary"
          heading={t(`${selectedForm}:stepper:step2`)}
          className="rectification-summary-fine-details">
          <InfoContainer />
        </Accordion>
      </div>
    </>
  );
};

export default RectificationSummary;
