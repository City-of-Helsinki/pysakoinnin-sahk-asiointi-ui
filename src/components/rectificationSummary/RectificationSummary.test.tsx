import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { axe } from 'jest-axe';
import RectificationSummary from './RectificationSummary';
import store from '../../store';
import { Provider } from 'react-redux';
import { configureStore, createSlice } from '@reduxjs/toolkit';

describe('Component', () => {
  it('matches snapshot', async () => {
    const formContentSliceMock = createSlice({
      name: 'formContent',
      initialState: {
        formSubmitted: false,
        selectedForm: 'parking-fine',
        submitDisabled: true,
        formValues: {
          invoiceNumber: '',
          refNumber: '',
          regNumber: '',
          relation: 'driver',
          poaFile: { name: 'test.pdf', size: 12345, type: 'application/pdf' },
          attachments: [{ name: 'test2.jpg', size: 50100, type: 'image/jpeg' }],
          toSeparateEmail: false,
          newEmailAddress: '',
          newEmailConfirm: '',
          address: 'Elim√§enkatu 5',
          zipCode: '00100',
          city: 'Helsinki',
          countryCode: '+358',
          phone: '401234567',
          IBAN: 'FI9780001700903330',
          rectificationContent:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse bibendum condimentum mi, vitae efficitur mi vulputate at. Aliquam porttitor tincidunt ex non fermentum. Fusce consequat imperdiet augue ut pulvinar. Praesent sollicitudin nulla non lacus tristique, sed faucibus urna viverra. Nullam pretium velit lorem. Maecenas porttitor molestie.',
          deliveryDecision: 'toParkingService'
        }
      },
      reducers: {}
    });

    const userSliceMock = createSlice({
      name: 'userProfile',
      initialState: {
        name: 'Test User',
        email: 'test.user@test.fi',
        SSN: '123456-789A'
      },
      reducers: {}
    });

    const store = configureStore({
      reducer: {
        formContent: formContentSliceMock.reducer,
        userProfile: userSliceMock.reducer
      }
    });
    const { container } = render(
      <Provider store={store}>
        <RectificationSummary />
      </Provider>
    );
    await waitFor(() => expect(container).toMatchSnapshot());
  });

  it('passes A11Y checks', async () => {
    const { container } = render(
      <Provider store={store}>
        <RectificationSummary />
      </Provider>
    );

    expect(await axe(container)).toHaveNoViolations();
  });
});
