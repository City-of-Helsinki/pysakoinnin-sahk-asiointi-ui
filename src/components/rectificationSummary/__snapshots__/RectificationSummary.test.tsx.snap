// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
<div>
  <h2
    class="rectification-summary-title"
  >
    Oikaisuvaatimuksen tiedot
  </h2>
  <div
    class="rectification-summary-container"
  >
    <div
      class="rectification-summary-details"
    >
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="relation"
        >
          Teen oikaisuvaatimuksen
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="relation"
            readonly=""
            type="text"
            value="Ajoneuvon kuljettajana"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="name"
        >
          Nimi
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="name"
            readonly=""
            type="text"
            value="Test User"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="ssn"
        >
          Henkilötunnus
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="ssn"
            readonly=""
            type="text"
            value="123456-789A"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="rectification-address"
        >
          Lähiosoite
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="rectification-address"
            readonly=""
            type="text"
            value="Elimäenkatu 5"
          />
        </div>
      </div>
      <div
        class="rectification-summary-subgrid"
      >
        <div
          class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
        >
          <label
            class="FieldLabel-module_label__1zrXK "
            for="zipcode"
          >
            Postinumero
          </label>
          <div
            class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
          >
            <input
              class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
              id="zipcode"
              readonly=""
              type="text"
              value="00100"
            />
          </div>
        </div>
        <div
          class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
        >
          <label
            class="FieldLabel-module_label__1zrXK "
            for="city"
          >
            Postitoimipaikka
          </label>
          <div
            class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
          >
            <input
              class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
              id="city"
              readonly=""
              type="text"
              value="Helsinki"
            />
          </div>
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="email"
        >
          Sähköpostiosoite
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="email"
            readonly=""
            type="text"
            value="test.user@test.fi"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="phone"
        >
          Puhelinnumero
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="phone"
            readonly=""
            type="text"
            value="+358401234567"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="IBAN"
        >
          Pankkitilinumero (IBAN)
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="IBAN"
            readonly=""
            type="text"
            value="FI9780001700903330"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="deliveryDecision"
        >
          Haluan päätöksen
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="deliveryDecision"
            readonly=""
            type="text"
            value="Pysäköinnin asiointikansiooni"
          />
        </div>
      </div>
    </div>
    <div
      class="rectification-summary-contents"
    >
      <div
        class="rectification-summary-rectification-content"
      >
        <label
          class="text-label"
          for="rectification-content"
        >
          Oikaisuvaatimuksen sisältö
        </label>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse bibendum condimentum mi, vitae efficitur mi vulputate at. Aliquam porttitor tincidunt ex non fermentum. Fusce consequat imperdiet augue ut pulvinar. Praesent sollicitudin nulla non lacus tristique, sed faucibus urna viverra. Nullam pretium velit lorem. Maecenas porttitor molestie.
        </p>
      </div>
      <div>
        <label
          class="text-label"
        >
          Liitetiedostot
        </label>
        <ul
          class="file-list"
        >
          <li
            class="file-list-item"
          >
            <svg
              aria-hidden="true"
              class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
              role="img"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <path
                  d="M0 0h24v24H0z"
                />
                <path
                  d="M20 2v20H4V2h16zm-4 13c-.86 0-1.284.222-2.12 1.036l-.173.171C12.411 17.503 11.583 18 10 18c-1.5 0-2.322-.446-3.506-1.595L6 15.92V20h12v-4.078l-.042-.04C17.226 15.193 16.803 15 16 15zm2-11H6l.001 9.44c.502.243.928.59 1.51 1.159l.196.194C8.661 15.747 9.083 16 10 16c.86 0 1.284-.222 2.12-1.036l.173-.171C13.589 13.497 14.417 13 16 13c.81 0 1.422.13 2 .43V4zm-8 2a3 3 0 110 6 3 3 0 010-6zm0 2a1 1 0 100 2 1 1 0 000-2z"
                  fill="currentColor"
                />
              </g>
            </svg>
            <div
              class="file-list-item-title"
            >
              <span
                class="file-list-item-name"
              >
                test2.jpg
              </span>
              <span
                class="file-list-item-size"
              >
                (
                49 KB
                )
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div>
        <label
          class="text-label"
        >
          Valtakirja
        </label>
        <div
          class="file-list-item"
        >
          <svg
            aria-hidden="true"
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <rect
                height="24"
                width="24"
              />
              <path
                d="M15,2 L20,7 L20,22 L4,22 L4,2 L15,2 Z M14,4 L14,8 L18,8 L14,4 Z M14.0082581,4 L18,8 L18,20 L6,20 L6,4 L14.0082581,4 Z M17,16 L7,16 L7,18 L17,18 L17,16 Z M17,12 L7,12 L7,14 L17,14 L17,12 Z M13,8 L7,8 L7,10 L13,10 L13,8 Z"
                fill="currentColor"
              />
            </g>
          </svg>
          <div
            class="file-list-item-title"
          >
            <span
              class="file-list-item-name"
            >
              test.pdf
            </span>
            <span
              class="file-list-item-size"
            >
              (
              12 KB
              )
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="hide-on-print"
  >
    <div
      class="custom-accordion-container"
    >
      <button
        class="Button-module_button__1msFE button_hds-button__2A0je Button-module_supplementary__3YKiS button_hds-button--supplementary__GcHcV custom-accordion custom-accordion-header"
        type="button"
      >
        <span
          class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
        >
          Pysäköintivirhemaksun tiedot
        </span>
        <div
          aria-hidden="true"
          class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
        >
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_xl__1M6cO icon_hds-icon--size-xl__1Jes8"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
                fill="currentColor"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div
    aria-hidden="true"
    class="show-on-print"
  >
    <div
      class="info-container"
    >
      <div
        class="summary-body"
      >
        <div
          class="summary-container"
          data-testid="parkingFineSummary"
        >
          <h2
            aria-hidden="true"
            class="show-on-print"
          >
            Pysäköintivirhemaksun tiedot
          </h2>
          <div
            class="info-field"
          >
            <label>
              Virheen ajankohta
            </label>
            <p>
              1.11.2019 09:41 - 1.11.2019 09:43
            </p>
          </div>
          <div
            class="info-field"
          >
            <label>
              Asianumero
            </label>
            <p>
              Esim. 12345678
            </p>
          </div>
          <hr />
          <div
            class="info-field"
          >
            <label>
              Lähiosoite
            </label>
            <p>
              Näkinkuja 1c
            </p>
          </div>
          <div
            class="info-field"
          >
            <label>
              Lisätietoja
            </label>
            <p>
              Kääntymispaikka
            </p>
          </div>
          <hr />
          <div
            class="info-field"
          >
            <label>
              Virhe
            </label>
            <p>
              Pysäköinti kielletty liikennemerkin noudattamatta jättäminen. TLL 3 § 1 mom, TL 16 § merkki 372
            </p>
          </div>
          <div
            class="info-field"
          >
            <label>
              Lisätieto
            </label>
            <p>
              Liikennemerkistä n. 3 m
            </p>
          </div>
          <div
            class="info-field"
          >
            <label>
              Virhe
            </label>
            <p>
              Pysäköinti kielletty liikennemerkin noudattamatta jättäminen. TLL 3 § 1 mom, TL 16 § merkki 372
            </p>
          </div>
          <div
            class="info-field"
          >
            <label>
              Lisätieto
            </label>
            <p>
              Liikennemerkistä n. 3 m
            </p>
          </div>
          <hr />
          <div
            class="wide-field"
          >
            <label>
              Pysäköintivirheen lisätiedot
            </label>
            <p>
              Lyökämme käsi kätehen, sormet sormien lomahan, lauloaksemme hyviä, parahia pannaksemme, kuulla noien kultaisien, tietä mielitehtoisien, nuorisossa nousevassa, kansassa kasuavassa: noita saamia sanoja, virsiä virittämiä vyöltä vanhan Väinämöisen, alta ahjon Ilmarisen, päästä kalvan Kaukomielen. 294 merkkiä
            </p>
          </div>
          <hr />
          <div
            class="info-field sum-field"
          >
            <label>
              Pysäköintivirhemaksun summa
            </label>
            <p>
              80,00 EUR
            </p>
          </div>
          <div
            class="info-field"
          >
            <label>
              Eräpäivä
            </label>
            <p>
              2.12.2019
            </p>
          </div>
          <hr />
          <div
            class="barcode-container wide-field"
          >
            <label
              class="barcode-label"
            >
              Virtuaaliviivakoodi
            </label>
            <p
              class="barcode"
            >
              430123730001230560012400000000000000000100018714210302
            </p>
            <button
              class="Button-module_button__1msFE button_hds-button__2A0je Button-module_secondary__1nABp button_hds-button--secondary__1NOWS barcode-button"
              data-testid="copy-to-clipboard"
              type="button"
            >
              <div
                aria-hidden="true"
                class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
              >
                <svg
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <rect
                      height="24"
                      width="24"
                    />
                    <path
                      d="M6,10 L6,12 L5,12 L5,18 L12,18 L12,17 L14,17 L14,19 C14,19.5522847 13.5522847,20 13,20 L4,20 C3.44771525,20 3,19.5522847 3,19 L3,11 C3,10.4477153 3.44771525,10 4,10 L6,10 Z M20,4 C20.5522847,4 21,4.44771525 21,5 L21,15 C21,15.5522847 20.5522847,16 20,16 L8,16 C7.44771525,16 7,15.5522847 7,15 L7,5 C7,4.44771525 7.44771525,4 8,4 L20,4 Z M19,6 L9,6 L9,14 L19,14 L19,6 Z"
                      fill="currentColor"
                    />
                  </g>
                </svg>
              </div>
              <span
                class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
              >
                Kopioi virtuaaliviivakoodi
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="Card-module_card__1WbKx card_hds-card--border__c2dBc custom-theme-1 vehicle-details-container"
        data-testid="carInfoCard"
        role="region"
      >
        <h2
          class="vehicle-details-header"
        >
          Ajoneuvon tiedot
        </h2>
        <div>
          <label>
            Ajoneuvon rekisteritunnus
          </label>
          <p>
            Esim. ABC-123
          </p>
        </div>
        <div>
          <label>
            Ajoneuvolaji
          </label>
          <p>
            Henkilöauto M1
          </p>
        </div>
        <div>
          <label>
            Merkki
          </label>
          <p>
            Citroen
          </p>
        </div>
        <div>
          <label>
            Malli
          </label>
          <p>
            2D C2 HATCHBACK 1.4l-JMKFVC/232
          </p>
        </div>
        <div>
          <label>
            Väri
          </label>
          <p>
            Punainen
          </p>
        </div>
        <div
          class="imageViewer-preview-container"
        >
          <input
            alt="Avaa kuvankatselu"
            class="imageViewer-preview-image"
            data-testid="clickable-image"
            formmethod="dialog"
            src="https://via.placeholder.com/600.png"
            style="height: 0px;"
            type="image"
            value="0"
          />
          <input
            alt="Avaa kuvankatselu"
            class="imageViewer-preview-image"
            data-testid="clickable-image"
            formmethod="dialog"
            src="https://via.placeholder.com/600x1200.png"
            style="height: 0px;"
            type="image"
            value="1"
          />
          <input
            alt="Avaa kuvankatselu"
            class="imageViewer-preview-image"
            data-testid="clickable-image"
            formmethod="dialog"
            src="https://via.placeholder.com/1200x800.png"
            style="height: 0px;"
            type="image"
            value="2"
          />
          <input
            alt="Avaa kuvankatselu"
            class="imageViewer-preview-image"
            data-testid="clickable-image"
            formmethod="dialog"
            src="https://via.placeholder.com/600x300.png"
            style="height: 0px;"
            type="image"
            value="3"
          />
          <input
            alt="Avaa kuvankatselu"
            class="imageViewer-preview-image"
            data-testid="clickable-image"
            formmethod="dialog"
            src="https://via.placeholder.com/900x800.png"
            style="height: 0px;"
            type="image"
            value="4"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
